Enhancements

sonokoh opened the pull request:
- enhanced add/edit features
- added drag and drop for sorting tasks
- added time support

cjshay commented on ui/src/components/todo-list/QTodoItemEditDialog.js:
```diff
@@ -1,73 +1,64 @@
+    setData (item) {
+      this.internalValue = item
+    },
+    $_renderDialogContent (h) {
+      return [
+        h(QCard, {
+          staticClass: 'q-todo-item-edit-popup full-width'
+        }, [
+          this.title && h(QCardSection, { staticClass: 'text-h6' }, this.title),
+          h(QCardSection, {
+            staticClass: 'q-pt-none'
+          }, [
+            h(QTodoItemEditForm, {
+              ref: 'form',
+              props: {
+                ...this.$props,
+                value: this.internalValue
+              },
+              on: {
+                submit: (val) => {
+                  this.$_submit(val)
+                },
+                cancel: () => {
+                  this.$_cancel()
+                }
+              }
+            })
+          ])
+        ])
+      ]
     }
   },
```
This function is a bit hard to read, maybe if you defined some of these as variables and passed them in that way? It's currently triple nested function-wise and hard to visually see where functions are starting / ending

This formatting isn't much better but easier for me to read personally (attached)
[todo-list-2.zip](https://github.com/Source-Digital/quasar-app-extension-tasks/files/5644416/todo-list-2.zip)

sonokoh replied:
Yeah I agree that render function is hard to read in general.  I personally still prefer formatting below: 
```
h(Component, {
  props: {}
},[
    child1,
    child2
])
```

as the above gets rendered as:

```
<Component v-on="props">
  <child1 />
  <child2 />
</Component>
```
I created local variable so there's less complex nesting. I'll be careful going forward. Ty for the feedback:)

cjshay reviewed (commented):
Low priority / Up to your discretion
1. Are you in control of the quickness of the transition between hour selection / minutes selection? It's a bit quick / jarring.
2. I really like the animation / functionality to remove the date by clicking it ==> unselecting it. What do you think about also adding a remove date / x button, for the person who might not realize clicking on the day on the calendar removes the date?
3. In http://localhost:8080/demo_-q-todo-list-edit, the edit needs some padding or a header in between the top of the input and the top of the edit modal (attached)
<img width="1433" alt="Screen Shot 2020-12-04 at 12 53 52 PM" src="https://user-images.githubusercontent.com/17301440/101197615-5b1eb000-3630-11eb-8e32-3e1bb5bd70ca.png">
4. How set in stone is this url routing schema? having a "_" and then a "-" seems a bit odd ex: demo_-q-todo-list-edit

sonokoh commented:
Hi thanks for your feedbacks! 
> 1. Are you in control of the quickness of the transition between hour selection / minutes selection? It's a bit quick / jarring.
Unfortunately not, I'm using Quasar's QTime component as is. https://quasar.dev/vue-components/time#QTime-API

> 2. I really like the animation / functionality to remove the date by clicking it ==> unselecting it. What do you think about also adding a remove date / x button, for the person who might not realize clicking on the day on the calendar removes the date?
Great feedback! I added clear button.

> 3. In http://localhost:8080/demo_-q-todo-list-edit, the edit needs some padding or a header in between the top of the input and the top of the edit modal (attached)
> 4. How set in stone is this url routing schema? having a "_" and then a "-" seems a bit odd ex: demo_-q-todo-list-edit
These demo pages are development only, and will not be part of published modules so we don't have to worry too much -- but it was bugging me as well so I fixed it:) 
These URIs are auto generated by Quasar's predefined routing mechanism for dev mode. It automatically detects page components in ui/dev/src/pages and generates slug uri from their names.
